back/app/services/multiloc_service.rb

```diff
@@ -5,12 +5,12 @@ def initialize(app_configuration: nil)
     @app_configuration = app_configuration
   end
 
-  def t(translations, user = nil)
+  def t(translations, preferred_locale = nil)
     return nil unless translations
 
     locales = app_configuration.settings('core', 'locales')
-    user_locale = user&.locale || I18n.locale.to_s
-    result = ([user_locale] + locales + translations.keys).each do |locale|
+    preferred_locale ||= I18n.locale.to_s
+    result = ([preferred_locale] + locales + translations.keys).each do |locale|
```

Comment:
Unrelated to your code change (which I like), but opportunistically: Reading this I'm wondering whether it wouldn't be worth it to refactor this so we don't call app_configuration in the case where we have a preferred_locale.

The repeated queries to app_configuration are a significant slowdown factor in our BE code, and this small change could maybe speed us up in many places?

Reply:
Good point! I'll rework that part.

> Sure you've covered all relevant cases of service.t?

I tried my best :)